components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -28,6 +29,15 @@ class Session(
             notifyObservers { onUrlChanged() }
         }
 
+    /**
+     * The progress loading the current URL.
+     */
+    var progress: Int = 0
+        set(value) {
+            field = value
+            notifyObservers { onProgress() }
+        }
+
```

Comment:
Can we add a loading boolean too? A lot of things just need to know whether we are currently loading a page or not. With that we avoid that (1) we need to check progress == 0 / 100 and (2) do not run code interesting in loading state changes for every percentage update.

Reply:
OK, I will add a boolean loading state to the session.